papas/simulation/Propagator.h

```diff
@@ -17,19 +17,26 @@ class Propagator {
    */
   Propagator(std::shared_ptr<const Field> field) : m_field(field){};
 
+  virtual ~Propagator() = default;
+
   /**
    Propagate particle to the selected cylinder and store the point where the particle crossed the cylinder
    @param[in] ptc particle that is to be propagated
    @param[in] cyl cylinder to which the particle is to be propagated.
    @param[in] field magnitude of magnetic field (used only for charged particles, defaults to zero if not set)
    */
-  virtual void propagateOne(Particle& ptc, const SurfaceCylinder& cyl) const = 0;
+  virtual void propagateOne(const Particle& ptc, const SurfaceCylinder& cyl) const = 0;
```

Comment:
Yes, it can (the particle contains the shared_pointer to the path and the shared_pointer is const).
There is more discussion of this in Design of Particle and Path  Issue #32:-
https://github.com/HEP-FCC/papas/issues/32